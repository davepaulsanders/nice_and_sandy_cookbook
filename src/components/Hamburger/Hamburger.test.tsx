import HamburgerMenu from "./Hamburger";
import { expect, test, vi } from "vitest";
import { render } from "@testing-library/react";

import "@testing-library/dom";
test("<Hamburger/> renders open", () => {
	const { container } = render(
		<HamburgerMenu isOpen={true} toggleOpen={vi.fn()}/>
	);
	expect(container).toMatchSnapshot();
});
test("<Hamburger/> renders closed", () => {
	const { container } = render(
		<HamburgerMenu isOpen={false} toggleOpen={vi.fn()}/>
	);
	expect(container).toMatchSnapshot();
});
