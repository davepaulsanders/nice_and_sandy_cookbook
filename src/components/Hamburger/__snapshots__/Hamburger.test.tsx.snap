// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Hamburger/> renders closed 1`] = `
<div>
  <button
    class="false fixed flex flex-col justify-center z-[1001] items-center w-[50px] h-[50px] top-5 right-10"
  >
    <span
      class="bg-lightg block transition-all duration-300 ease-out h-0.5 w-6 rounded-sm -translate-y-0.5"
    />
    <span
      class="bg-lightg block transition-all duration-300 ease-out h-0.5 w-6 rounded-sm my-0.5 opacity-100"
    >
       
    </span>
    <span
      class="bg-lightg block transition-all duration-300 ease-out h-0.5 w-6 rounded-sm translate-y-0.5"
    >
       
    </span>
  </button>
</div>
`;

exports[`<Hamburger/> renders open 1`] = `
<div>
  <button
    class="false fixed flex flex-col justify-center z-[1001] items-center w-[50px] h-[50px] top-5 right-10"
  >
    <span
      class="bg-lightg block transition-all duration-300 ease-out h-0.5 w-6 rounded-sm rotate-45 translate-y-1"
    />
    <span
      class="bg-lightg block transition-all duration-300 ease-out h-0.5 w-6 rounded-sm my-0.5 opacity-0"
    >
       
    </span>
    <span
      class="bg-lightg block transition-all duration-300 ease-out h-0.5 w-6 rounded-sm -rotate-45 -translate-y-1"
    >
       
    </span>
  </button>
</div>
`;
